compress.hh

```diff
@@ -97,6 +97,8 @@ public:
     static thread_local const ptr_type deflate;
 
     static const sstring namespace_prefix;
+private:
+    virtual ptr_type replace(const opt_getter&) const;
```

Comment:
Please write a comment specifying what this function does, especially since it's a virtual interface

And in the default implementation, explain why returning `nullptr` makes sense

Reply:
Right